Issue continue training on tansformer based NER

I'm using the nightly version, I have successfully trained a transformer based NER model and saved it;  now I'm trying to resume training on it. 

Firstly, I'm not sure if I have set up the config file correctly, the relevant part looks like this:

```
[components]

[components.ner]
# This is the path to my trained model
source='best-model'

[components.ner.model]
@architectures = "spacy.TransitionBasedParser.v1"
state_type = "ner"
extra_state_tokens = false
hidden_width = 64
maxout_pieces = 2
use_upper = false
nO = null

[components.ner.model.tok2vec]
@architectures = "spacy-transformers.TransformerListener.v1"
grad_factor = 1.0
pooling = {"@layers":"reduce_mean.v1"}

[components.transformer]
# This is the path to my trained model
source='best-model'

[components.transformer.model]
@architectures = "spacy-transformers.TransformerModel.v1"
name = "dccuchile/bert-base-spanish-wwm-uncased"

[components.transformer.model.get_spans]
@span_getters = "spacy-transformers.strided_spans.v1"
window = 128
stride = 96

[components.transformer.model.tokenizer_config]
use_fast = true
```

I'm getting this error message:

```
2020-10-29 14:36:11.541313: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.10.1
ℹ Using GPU: 0

=========================== Initializing pipeline ===========================
Set up nlp object from config
Pipeline: ['transformer', 'ner']
Resuming training for: ['ner', 'transformer']
Created vocabulary
Finished initializing nlp object
Initialized pipeline components: []
✔ Initialized pipeline

============================= Training pipeline =============================
ℹ Pipeline: ['transformer', 'ner']
ℹ Initial learn rate: 0.0
E    #       LOSS TRANS...  LOSS NER  ENTS_F  ENTS_P  ENTS_R  SCORE 
---  ------  -------------  --------  ------  ------  ------  ------
⚠ Aborting and saving the final best model. Encountered exception: CUDA
out of memory. Tried to allocate 94.00 MiB (GPU 0; 15.75 GiB total capacity;
13.81 GiB already allocated; 78.88 MiB free; 14.34 GiB reserved in total by
PyTorch)
✔ Saved pipeline to output directory
model_t2/model-last
Traceback (most recent call last):
  File "/usr/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/usr/local/lib/python3.6/dist-packages/spacy/__main__.py", line 4, in <module>
    setup_cli()
  File "/usr/local/lib/python3.6/dist-packages/spacy/cli/_util.py", line 65, in setup_cli
    command(prog_name=COMMAND)
  File "/usr/local/lib/python3.6/dist-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.6/dist-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.6/dist-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.6/dist-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/typer/main.py", line 497, in wrapper
    return callback(**use_params)  # type: ignore
  File "/usr/local/lib/python3.6/dist-packages/spacy/cli/train.py", line 59, in train_cli
    train(nlp, output_path, use_gpu=use_gpu, stdout=sys.stdout, stderr=sys.stderr)
  File "/usr/local/lib/python3.6/dist-packages/spacy/training/loop.py", line 105, in train
    raise e
  File "/usr/local/lib/python3.6/dist-packages/spacy/training/loop.py", line 85, in train
    for batch, info, is_best_checkpoint in training_step_iterator:
  File "/usr/local/lib/python3.6/dist-packages/spacy/training/loop.py", line 201, in train_while_improving
    score, other_scores = evaluate()
  File "/usr/local/lib/python3.6/dist-packages/spacy/training/loop.py", line 253, in evaluate
    scores = nlp.evaluate(dev_examples)
  File "/usr/local/lib/python3.6/dist-packages/spacy/language.py", line 1312, in evaluate
    docs = list(docs)
  File "/usr/local/lib/python3.6/dist-packages/spacy/util.py", line 1363, in _pipe
    yield from proc.pipe(docs, **kwargs)
  File "spacy/pipeline/transition_parser.pyx", line 170, in pipe
  File "/usr/local/lib/python3.6/dist-packages/spacy/util.py", line 1322, in minibatch
    batch = list(itertools.islice(items, int(batch_size)))
  File "/usr/local/lib/python3.6/dist-packages/spacy/util.py", line 1363, in _pipe
    yield from proc.pipe(docs, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/spacy_transformers/pipeline_component.py", line 173, in pipe
    self.set_annotations(subbatch, self.predict(subbatch))
  File "/usr/local/lib/python3.6/dist-packages/spacy_transformers/pipeline_component.py", line 189, in predict
    activations = self.model.predict(docs)
  File "/usr/local/lib/python3.6/dist-packages/thinc/model.py", line 312, in predict
    return self._func(self, X, is_train=False)[0]
  File "/usr/local/lib/python3.6/dist-packages/spacy_transformers/layers/transformer_model.py", line 111, in forward
    tensors, bp_tensors = transformer(token_data, is_train)
  File "/usr/local/lib/python3.6/dist-packages/thinc/model.py", line 288, in __call__
    return self._func(self, X, is_train=is_train)
  File "/usr/local/lib/python3.6/dist-packages/thinc/layers/pytorchwrapper.py", line 79, in forward
    Ytorch, torch_backprop = model.shims[0](Xtorch, is_train)
  File "/usr/local/lib/python3.6/dist-packages/thinc/shims/pytorch.py", line 29, in __call__
    return self.predict(inputs), lambda a: ...
  File "/usr/local/lib/python3.6/dist-packages/thinc/shims/pytorch.py", line 38, in predict
    outputs = self._model(*inputs.args, **inputs.kwargs)
  File "/usr/local/lib/python3.6/dist-packages/torch/nn/modules/module.py", line 722, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/transformers/modeling_bert.py", line 762, in forward
    output_hidden_states=output_hidden_states,
  File "/usr/local/lib/python3.6/dist-packages/torch/nn/modules/module.py", line 722, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/transformers/modeling_bert.py", line 439, in forward
    output_attentions,
  File "/usr/local/lib/python3.6/dist-packages/torch/nn/modules/module.py", line 722, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/transformers/modeling_bert.py", line 388, in forward
    intermediate_output = self.intermediate(attention_output)
  File "/usr/local/lib/python3.6/dist-packages/torch/nn/modules/module.py", line 722, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/transformers/modeling_bert.py", line 333, in forward
    hidden_states = self.intermediate_act_fn(hidden_states)
  File "/usr/local/lib/python3.6/dist-packages/torch/nn/functional.py", line 1369, in gelu
    return torch._C._nn.gelu(input)
**RuntimeError: CUDA out of memory. Tried to allocate 94.00 MiB (GPU 0; 15.75 GiB total capacity; 13.81 GiB already allocated; 78.88 MiB free; 14.34 GiB reserved in total by PyTorch)**
```

I understand the message is telling me I'm out of memory, but it seems weird that I'm able to train from scratch with no issues but getting this error when trying to resume training on the saved model. Any help is appreciated.

## Your Environment
- **spaCy version:** 3.0.0rc2
- **Platform:** Linux-4.19.112+-x86_64-with-Ubuntu-18.04-bionic
- **Python version:** 3.6.9
- **Pipelines:** es_core_news_md (3.0.0a0), es_dep_news_trf (3.0.0a0)

--- a/spacy/language.py
+++ b/spacy/language.py
@@ -1295,27 +1295,28 @@ class Language:
             kwargs = dict(scorer_cfg)
             kwargs.setdefault("nlp", self)
             scorer = Scorer(**kwargs)
-        texts = [eg.reference.text for eg in examples]
-        docs = [eg.predicted for eg in examples]
+        # reset annotation in predicted docs and time tokenization
         start_time = timer()
-        # tokenize the texts only for timing purposes
         if not hasattr(self.tokenizer, "pipe"):
-            _ = [self.tokenizer(text) for text in texts]  # noqa: F841
+            for eg in examples:
+                eg.predicted = self.tokenizer(eg.reference.text)
         else:
-            _ = list(self.tokenizer.pipe(texts))  # noqa: F841
+            for doc, eg in zip(
+                self.tokenizer.pipe(eg.reference.text for eg in examples), examples
+            ):
+                eg.predicted = doc
+        # apply all pipeline components
+        docs = [eg.predicted for eg in examples]
         for name, pipe in self.pipeline:
             kwargs = component_cfg.get(name, {})
             kwargs.setdefault("batch_size", batch_size)
             docs = _pipe(docs, pipe, kwargs)
-        # iterate over the final generator
-        if len(self.pipeline):
-            docs = list(docs)
+        # iterate over final generator
+        for doc in docs:
+            pass
         end_time = timer()
-        for i, (doc, eg) in enumerate(zip(docs, examples)):
-            util.logger.debug(doc)
-            eg.predicted = doc
         results = scorer.score(examples)
-        n_words = sum(len(doc) for doc in docs)
+        n_words = sum(len(eg.predicted) for eg in examples)
         results["speed"] = n_words / (end_time - start_time)
         return results
 

--- a/spacy/language.py
+++ b/spacy/language.py
@@ -1286,6 +1286,7 @@ class Language:
 
         DOCS: https://nightly.spacy.io/api/language#evaluate
         """
+        examples = list(examples)
         validate_examples(examples, "Language.evaluate")
         if component_cfg is None:
             component_cfg = {}

--- a/spacy/language.py
+++ b/spacy/language.py
@@ -1306,18 +1306,13 @@ class Language:
             ):
                 eg.predicted = doc
         # apply all pipeline components
-        docs = None
         for name, pipe in self.pipeline:
             kwargs = component_cfg.get(name, {})
             kwargs.setdefault("batch_size", batch_size)
-            if docs is None:
-                docs = _pipe((eg.predicted for eg in examples), pipe, kwargs)
-            else:
-                docs = _pipe(docs, pipe, kwargs)
-        # iterate over final generator
-        if docs is not None:
-            for doc in docs:
-                pass
+            for doc, eg in zip(
+                _pipe((eg.predicted for eg in examples), pipe, kwargs), examples
+            ):
+                eg.predicted = doc
         end_time = timer()
         results = scorer.score(examples)
         n_words = sum(len(eg.predicted) for eg in examples)

--- a/spacy/language.py
+++ b/spacy/language.py
@@ -1311,7 +1311,7 @@ class Language:
             kwargs = component_cfg.get(name, {})
             kwargs.setdefault("batch_size", batch_size)
             if docs is None:
-                docs = _pipe(eg.predicted for eg in examples, pipe, kwargs)
+                docs = _pipe((eg.predicted for eg in examples), pipe, kwargs)
             else:
                 docs = _pipe(docs, pipe, kwargs)
         # iterate over final generator

--- a/spacy/language.py
+++ b/spacy/language.py
@@ -1306,11 +1306,14 @@ class Language:
             ):
                 eg.predicted = doc
         # apply all pipeline components
-        docs = [eg.predicted for eg in examples]
+        docs = None
         for name, pipe in self.pipeline:
             kwargs = component_cfg.get(name, {})
             kwargs.setdefault("batch_size", batch_size)
-            docs = _pipe(docs, pipe, kwargs)
+            if docs is None:
+                docs = _pipe(eg.predicted for eg in examples, pipe, kwargs)
+            else:
+                docs = _pipe(docs, pipe, kwargs)
         # iterate over final generator
         for doc in docs:
             pass

--- a/spacy/language.py
+++ b/spacy/language.py
@@ -1315,8 +1315,9 @@ class Language:
             else:
                 docs = _pipe(docs, pipe, kwargs)
         # iterate over final generator
-        for doc in docs:
-            pass
+        if docs is not None:
+            for doc in docs:
+                pass
         end_time = timer()
         results = scorer.score(examples)
         n_words = sum(len(eg.predicted) for eg in examples)

--- a/spacy/language.py
+++ b/spacy/language.py
@@ -1298,14 +1298,8 @@ class Language:
             scorer = Scorer(**kwargs)
         # reset annotation in predicted docs and time tokenization
         start_time = timer()
-        if not hasattr(self.tokenizer, "pipe"):
-            for eg in examples:
-                eg.predicted = self.tokenizer(eg.reference.text)
-        else:
-            for doc, eg in zip(
-                self.tokenizer.pipe(eg.reference.text for eg in examples), examples
-            ):
-                eg.predicted = doc
+        for eg in examples:
+            eg.predicted = self.make_doc(eg.reference.text)
         # apply all pipeline components
         for name, pipe in self.pipeline:
             kwargs = component_cfg.get(name, {})

--- a/spacy/tests/test_language.py
+++ b/spacy/tests/test_language.py
@@ -53,7 +53,8 @@ def test_language_evaluate(nlp):
     annots = {"doc_annotation": {"cats": {"POSITIVE": 1.0, "NEGATIVE": 0.0}}}
     doc = Doc(nlp.vocab, words=text.split(" "))
     example = Example.from_dict(doc, annots)
-    nlp.evaluate([example])
+    scores = nlp.evaluate([example])
+    assert scores["speed"] > 0
 
     # Not allowed to call with just one Example
     with pytest.raises(TypeError):

--- a/spacy/tests/test_language.py
+++ b/spacy/tests/test_language.py
@@ -56,6 +56,10 @@ def test_language_evaluate(nlp):
     scores = nlp.evaluate([example])
     assert scores["speed"] > 0
 
+    # test with generator
+    scores = nlp.evaluate(eg for eg in [example])
+    assert scores["speed"] > 0
+
     # Not allowed to call with just one Example
     with pytest.raises(TypeError):
         nlp.evaluate(example)

--- a/spacy/training/loop.py
+++ b/spacy/training/loop.py
@@ -248,9 +248,8 @@ def create_evaluation_callback(
     weights = {key: value for key, value in weights.items() if value is not None}
 
     def evaluate() -> Tuple[float, Dict[str, float]]:
-        dev_examples = list(dev_corpus(nlp))
         try:
-            scores = nlp.evaluate(dev_examples)
+            scores = nlp.evaluate(dev_corpus(nlp))
         except KeyError as e:
             raise KeyError(Errors.E900.format(pipeline=nlp.pipe_names)) from e
         # Calculate a weighted sum based on score_weights for the main score.

